### PR TITLE
Test(Metadata): Remove catch from Setup()

### PR DESCRIPTION
**Pull request by chennes**
If an exception occurs while setting up Xerces-C, make the test framework handle it.

---

- [X]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466">pipeline status</a> for feature branch [PR_8466](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466). Pipeline [775402247](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775402247) was triggered at [b99b6b8](https://github.com/FreeCAD/FreeCAD/pull/8466/commits/b99b6b8ddda1cda1ca00a27e4dfa731617e12c1f). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466">pipeline status</a> for feature branch [PR_8466](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466). Pipeline [775462055](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775462055) was triggered at [dbff985](https://github.com/FreeCAD/FreeCAD/pull/8466/commits/dbff9858da24d887638b091e8bd3e74e74a15d86). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466">pipeline status</a> for feature branch [PR_8466](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466). Pipeline [775474027](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775474027) was triggered at [ce7545e](https://github.com/FreeCAD/FreeCAD/pull/8466/commits/ce7545e06b0dfaaadb8401602ad50bb1d242ef40). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466">pipeline status</a> for feature branch [PR_8466](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466). Pipeline [775479657](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775479657) was triggered at [3e39f04](https://github.com/FreeCAD/FreeCAD/pull/8466/commits/3e39f045085fbd88d88f089e5959907d3b401f2c). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466">pipeline status</a> for feature branch [PR_8466](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8466). Pipeline [775487770](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775487770) was triggered at [9133727](https://github.com/FreeCAD/FreeCAD/pull/8466/commits/9133727622351d0b91871d1ebd2954afc48c4ab7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: tests/src/App/Metadata.cpp [hunk 1]
```diff
@@ -147,19 +173,14 @@ TEST(VersionTest, VersionOperatorComparison){
...
-        }
+        xercesc_3_2::XMLPlatformUtils::Initialize();
     }
     void TearDown() override {
         xercesc_3_2::XMLPlatformUtils::Terminate();
     }
     std::string GivenSimpleMetadataXMLString()
     {
-        std::stringstream stream;
+        std::ostringstream stream;
```
Comment (chennes) [inline tests/src/App/Metadata.cpp:183]: @wwmayer do you have any idea why the linter is complaining about this line? There should be no need to do anything to initialize an `ostringstream`, as far as I am aware.

Comment (wwmayer) [reply tests/src/App/Metadata.cpp:183]: No idea what it is complaining about. Sometimes the tool just tells you nonsense. But what you could try is to write: `std::ostringstream stream{};`

**Pull request merged**